pkg/tuned/controller.go

```diff
@@ -1029,19 +1031,18 @@ func (c *Controller) changeSyncerTuneD(change Change) (synced bool, err error) {
 			// Cache the value written to tunedRecommendFile.
 			c.daemon.recommendedProfile = change.recommendedProfile
 			klog.V(1).Infof("recommended TuneD profile updated from %q to %q [inplaceUpdate=%v nodeRestart=%v]", prevRecommended, change.recommendedProfile, inplaceUpdate, change.nodeRestart)
-			changeRecommend = true
 
 			if change.deferredMode == util.DeferUpdate && !inplaceUpdate && c.daemon.recoveredRecommendedProfile == change.recommendedProfile {
```

Comment:
Is it possible to improve the comments/logging for this section?   I'm also a bit uneasy with the complex condition without a comment explaining in what state exactly we've currently landed in.  Ideally, the log message below should also explain it.

Reply:
added improvements, thanks for raising the point